### ISSUE TITLE
Is it possible to make `qAlign()` dependent on `Rbowtie2`?

### ISSUE THREAD

**Issue by gwangjinkim** [fmicompbio/QuasR#17]
`qAlign()` currently depends on `Rbowtie`. However, the index building procedure takes a lot of time (a whole afternoon and evening) and I didn't found a possibility to make it multi-cpu-driven, although one gives a `clObj` argument when calling `qAlign()`. Anyway, I am asking myself, why it does not depend on `Rbowtie2`.
Then the indexing could take place in a parallelized manner. And even better: One could pass the Bowtie2 index path as an argument. Would that be possible as a new feature?

**Comment by mbstadler** [comment]
`qAlign()` also supports `Rhisat2` for spliced (RNA-seq) alignments.

Supporting `Rbowtie2` would require a couple of adjustments to many parts in `QuasR`:  
- `bowtie2` is not an ungapped aligner (it allows for short indels), and there are multiple places in `QuasR` where the alignment end coordinate is currently calculated as the start coordinate plus the read length (for example in `qCount` and `qProfile` to check for overlapping reads, and in `qMeth` to access aligned bases in read and genome). Supporting `bowtie2` alignments would mean to use the CIGAR string to calculate alignment end coordiantes in all those cases.  
- `bowtie2` (similarly as `hisat2`) does not have a parameter to enforce suppresion of alignments where the read has more than `maxHits` possible best alignments. When using `Rhisat2` with `QuasR`, we work around that by first asking for `maxHits+1` reported alignments, and then filtering them to enforce `maxHits`. This would also have to be implemented for `bowtie2`  
- `qAlign` is not written very modularly: adding support for `Rbowtie2` would require addition of a number of specific functions to cover all use-cases (single read/paired-end alignments, spliced/unspliced/bisulfite alignments, and combinations thereof).  

In summary, it could be done, but it would take some work. 
We currently prioritize other tasks, especially since the index building has to be done only once.

We are happy to consider PRs and re-open this issue though, if you are willing to have a go at it.
